Traceback (most recent call last):
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/site-packages/jupyter_cache/executors/utils.py", line 56, in single_nb_execution
    record_timing=False,
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 1117, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/site-packages/nbclient/util.py", line 78, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/site-packages/nbclient/util.py", line 57, in just_run
    return loop.run_until_complete(coro)
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/asyncio/base_events.py", line 583, in run_until_complete
    return future.result()
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 559, in async_execute
    cell, index, execution_count=self.code_cells_executed + 1
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 862, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/Users/jeanwalrand/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 765, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# It is sometimes useful to sort a list.  Here, we sort the list A by decreasing value of the sine of the element.
def bySine(x):
    return math.sin(x)
A.sort(key = bySine, reverse = True)
A
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m<ipython-input-3-cca079e59b55>[0m in [0;36m<module>[0;34m[0m
[1;32m      2[0m [0;32mdef[0m [0mbySine[0m[0;34m([0m[0mx[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m      3[0m     [0;32mreturn[0m [0mmath[0m[0;34m.[0m[0msin[0m[0;34m([0m[0mx[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 4[0;31m [0mA[0m[0;34m.[0m[0msort[0m[0;34m([0m[0mkey[0m [0;34m=[0m [0mbySine[0m[0;34m,[0m [0mreverse[0m [0;34m=[0m [0;32mTrue[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      5[0m [0mA[0m[0;34m[0m[0;34m[0m[0m

[0;32m<ipython-input-3-cca079e59b55>[0m in [0;36mbySine[0;34m(x)[0m
[1;32m      1[0m [0;31m# It is sometimes useful to sort a list.  Here, we sort the list A by decreasing value of the sine of the element.[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m      2[0m [0;32mdef[0m [0mbySine[0m[0;34m([0m[0mx[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 3[0;31m     [0;32mreturn[0m [0mmath[0m[0;34m.[0m[0msin[0m[0;34m([0m[0mx[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      4[0m [0mA[0m[0;34m.[0m[0msort[0m[0;34m([0m[0mkey[0m [0;34m=[0m [0mbySine[0m[0;34m,[0m [0mreverse[0m [0;34m=[0m [0;32mTrue[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m      5[0m [0mA[0m[0;34m[0m[0;34m[0m[0m

[0;31mNameError[0m: name 'math' is not defined
NameError: name 'math' is not defined

